packages/app-config-writer/src/preview-config/prerequisites.ts

```diff
@@ -0,0 +1,75 @@
+ *
+ * @param basePath - base path to be used for the conversion
+ * @param fs - file system reference
+ * @param logger logger to report info to the user
+ * @returns indicator if the prerequisites are met
+ */
+export async function checkPrerequisites(basePath: string, fs: Editor, logger?: ToolsLogger): Promise<boolean> {
+    const packageJsonPath = join(basePath, 'package.json');
+    const packageJson = fs.readJSON(packageJsonPath) as Package | undefined;
+    let prerequisitesMet = true;
+
+    if (!packageJson) {
+        throw Error(`File 'package.json' not found at '${basePath}'`);
+    }
+
+    const sapui5BestpracticeBuildExists =
+        !!packageJson?.devDependencies?.['@sap/grunt-sapui5-bestpractice-build'] ||
+        !!packageJson?.dependencies?.['@sap/grunt-sapui5-bestpractice-build'];
+    if (sapui5BestpracticeBuildExists) {
+        logger?.error(
+            "A conversion from '@sap/grunt-sapui5-bestpractice-build' is not supported. Please migrate to UI5 CLI version 3.0.0 or higher first. See https://sap.github.io/ui5-tooling/v3/updates/migrate-v3/ for more information."
+        );
+        prerequisitesMet = false;
+    }
+
+    const ui5CliVersion = packageJson?.devDependencies?.['@ui5/cli'] ?? packageJson?.dependencies?.['@ui5/cli'] ?? '0';
+    if (parseInt(ui5CliVersion.split('.')[0], 10) < 3) {
+        logger?.error(
+            'UI5 CLI version 3.0.0 or higher is required to convert the preview to virtual files. See https://sap.github.io/ui5-tooling/v3/updates/migrate-v3/ for more information.'
+        );
+        prerequisitesMet = false;
+    }
+
+    const ui5MiddlewareMockserverExists =
+        !!packageJson?.devDependencies?.['@sap-ux/ui5-middleware-fe-mockserver'] ||
+        !!packageJson?.dependencies?.['@sap-ux/ui5-middleware-fe-mockserver'];
+    const cdsPluginUi5Exists =
+        !!packageJson?.devDependencies?.['cds-plugin-ui5'] || !!packageJson?.dependencies?.['cds-plugin-ui5'];
+    if (!ui5MiddlewareMockserverExists && !cdsPluginUi5Exists) {
+        logger?.error(
+            "A conversion from 'sap/ui/core/util/MockServer' is not supported. Please migrate to '@sap-ux/ui5-middleware-fe-mockserver' first (details see https://www.npmjs.com/package/@sap-ux/ui5-middleware-fe-mockserver)."
+        );
+        prerequisitesMet = false;
+    }
+
+    return prerequisitesMet;
+}
+
+/**
+ * Get the explicit approval form the user to do the conversion.
+ *
+ * @returns Explicit user approval to do the conversion.
+ */
+export async function getExplicitApprovalToAdjustFiles(): Promise<boolean> {
+    const question: PromptObject = {
+        type: 'confirm',
+        name: 'approval',
+        initial: false,
+        message:
+            'The converter will rename html files and delete js/ts files used for the existing preview and configure the usage of virtual files instead. Do you want to proceed with the conversion?'
+    };
+    return !!prompt([question]).approval;
```

Comment:
@heimwege, my review is not completely done, but I wanted to mention an issue I found during testing. The prompt did not wait for my user input, so I think we need to add `await` and `(` `)` here. Also I would use `Boolean()` instead of `!!`, I think it is more readable:

```suggestion
    return Boolean((await prompt([question])).approval);
```

Reply:
@Klaus-Keller the missing `await` was due to my refactoring - my bad. Did the change in f15564f8e548600231d7acbf7f06df6204e5a6ff